src/helpers/xml.js

```diff
@@ -21,20 +21,74 @@ export function encodeXML(str = '') {
  * Build a stringified OSM changeset
  * @param {string} [createdBy]
  * @param {string} [comment]
+ * @param {string} [optionalTags] Keys values to set tags
  * @return {string}
  */
-export function buildChangesetXml(createdBy = '', comment = '') {
+export function buildChangesetXml(
+  createdBy = '',
+  comment = '',
+  optionalTags = {}
+) {
+  const tags = Object.entries(optionalTags).map(entry => {
+    return `<tag k="${entry[0]}" v="${encodeXML(String(entry[1]))}"/>`;
+  });
   return `
     <osm>
       <changeset>
         <tag k="created_by" v="${encodeXML(createdBy)}"/>
         <tag k="created_by:library" v="OSM Request ${osmRequestVersion}"/>
         <tag k="comment" v="${encodeXML(comment)}"/>
+        ${tags.join(`\n        `)}
```

Comment:
This one is to allow custom tags to be added to changeset, like "host", "locale"... used by other editors. For pic4review, it would allow to keep track of mission URL for checks.